src/containers/Home/HomeWidgets/NotificationWidget.js

```diff
@@ -0,0 +1,283 @@
+// has the state changed hook
+const useCompare = (val) => {
+    const prevVal = usePrevious(val)
+    return prevVal !== val
+}
+
+// Helper hook
+const usePrevious = (value) => {
+    const ref = useRef();
+    useEffect(() => {
+        ref.current = value;
+    }, [value]);
+    return ref.current;
+}
+
+const createNotificationText = (text, icon, index) => {
+
+    return (
+        <View
+            style={{
+                flex: 1,
+                flexDirection: 'row',
+                width: "70%",
+                alignItems: 'center',
+                justifyContent: 'flex-start',
+                marginRight: 20
+            }}>
+            {icon}
+            <Text style={{ flexShrink: 1, marginLeft: 7 }}>
+                <Text style={{ fontSize: 12, color: Colors.primaryColor, fontWeight: 'bold', marginVertical: 5, }}>
+                    {typeof (text) === 'object' ? text[0] : ''}
+                </Text>
+                <Text style={{ fontSize: 12, color: "black", marginVertical: 5 }}>
+                    {typeof (text) === 'object' ? text[1] : text}
+                </Text>
+                <Text style={{ fontSize: 12, color: Colors.primaryColor, fontWeight: 'bold', marginVertical: 5, }}>
+                    {typeof (text) === 'object' ? text[2] : ''}
+                </Text>
+                <Text style={{ fontSize: 12, color: "black", marginVertical: 5 }}>
+                    {typeof (text) === 'object' ? text[3] : ''}
+                </Text>
+            </Text>
+        </View>
+    )
+}
+
+const getIcon = (type, index) => {
+
+    switch (type) {
+        case NOTIFICATION_ICON_VERUSID:
+            return (<VerusIdAtIcon
+                index={index}
+                width={20}
+                height={20}
+                marginLeft={7}
+                style={{
+                    alignSelf: 'center',
+                }} />);
+        case NOTIFICATION_ICON_ERROR:
+            return (<VerusIdErrorIcon
+                index={index}
+                width={23}
+                height={20}
+                marginLeft={6}
+                style={{
+                    alignSelf: 'center',
+                }} />);
+        case NOTIFICATION_ICON_TX:
+        default:
+            return (<ReceivedIcon
+                index={index}
+                width={20}
+                height={20}
+                marginLeft={7}
+                style={{
+                    alignSelf: 'center',
+                }} />);
+    }
+}
+
+const getNotifications = ({ directory } = notifications, acchash) => {
+
+    let tempNotificaions = [];
+    const keys = Object.keys(directory);
+
+    keys.forEach((uid, index) => {
+
+        if (directory[uid].acchash === acchash) {
+            if (directory[uid].type === NOTIFICATION_TYPE_DEEPLINK) {
+                const tempDeepLinkNotification = DeeplinkNotification.fromJson(directory[uid], processVerusId);
```

Comment:
Deeplink notifications are not verusid specific so we can't be calling processVerusId for all deeplink notifications as their onAction. There needs to be a way to process notification onActions and somehow link multiple different onAction functions to notifications here.